### ISSUE TITLE
[Bug] Not retrieving all groups of users

### ISSUE THREAD

**Issue by shark2k** [DirectoryTree/LdapRecord#557]
**Environment:**
 - LDAP Server Type: ActiveDirectory
 - PHP Version: 7.4
 - LdapRecord Version: 2.20.1

I am not 100% certain if this is a bug or working as intended but I decided to put this here (instead of in the discussions) in case it is a bug that can be resolved.

I have a small script that I wrote to get all the groups for a user.
The portion for LdapRecord is below (which is essentially the same as the example here, https://ldaprecord.com/docs/core/v1/active-directory/users/#getting-groups, with the exception that I am using FindByOrFail and looking at the username.

```php
<?php

use LdapRecord\Models\ActiveDirectory\User;

$user = User::findByOrFail('samaccountname', 'johndoe');

// Get immediate groups the user is apart of:
$groups = $user->groups()->get();

foreach ($groups as $group) {
    echo $group->getName();
}
```
The expected outcome of this is to get all the groups that the user is a member of, no matter what OU that group is from (we have many OUs for all the different countries we have subsidiaries in and in my case the groups for my users are either going to be in the US OU or our corporate OU).

At first I thought this was working, but then I realized that there were groups missing and the missing groups are any group that is not within the US OU.  The quickest group to notice, since all users have it, is the Domain Users group is not being pulled.  That group is a top level OU, right under domain.  If that's not clear what I mean is it is Domain->Users whereas the US OU is Domain->Corp->US.

To check this, I used my account which is a member of 68 groups and when I ran my script I only get 28 groups back.

I'm sure this will be a question, so I did a test using PowerShell and this line of code
```powershell
Get-ADPrincipalGroupMembership | select name
```

with the user that I am using to connect to AD with LdapRecord and in PowerShell I am returning all 68 groups.

I am not sure what additional information you would require for troubleshooting this, but I will provide as much information as possible if anything is requested.

Thanks,
-Shark2k

**Comment by stevebauman** [comment]
Hi @shark2k,

Do you have your `base_dn` set appropriately in your configuration to encompass your whole organization?

```php
$config = [
    'base_dn' => 'ou=Corporate,dc=local,dc=om' // <-- Scoped to only return results beneath "Corporate"

    'base_dn' => 'dc=local,dc=om' // <-- Scoped to entire domain
];
```

**Comment by shark2k** [comment]
Hi @stevebauman,

I should have went with my initial instinct and put this as a question under discussions.

You are correct in thinking my base_dn was set to search only the US OU.
I did that because I figured it would help with speed and also because generally I'm not concerned about anything outside of my OU.
I didn't realize it would have an impact on finding the groups that a user is a member of if they were outside.

I was going to ask a question but figured out how to test limiting the base DN for the command I mentioned in the OP.
I limited that to the same as I had in my script and I ended up with the same results (28 groups vs the 68 groups without the base DN set).
Based on that, it's working as expected and I shouldn't have opened this as a bug, so sorry about that.

Thanks for the quick response and pointing me in the right direction (again!).
-Shark2k
